File: app/src/main/java/it/liceoarzignano/bold/intro/BenefitPageAdapter.java
package it.liceoarzignano.bold.intro;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.util.SparseArray;
import android.view.ViewGroup;

class BenefitPageAdapter extends FragmentPagerAdapter {
    private final SparseArray<BenefitFragment> mFragments;

    BenefitPageAdapter(FragmentManager manager) {
        super(manager);
        mFragments = new SparseArray<>();
    }

    @Override
    public Fragment getItem(int position) {
        return new BenefitFragment().newInstance(position);
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        BenefitFragment fragment = (BenefitFragment) super.instantiateItem(container, position);
        mFragments.put(position, fragment);
        return fragment;
    }

    @Override
    public int getCount() {
        return 2;
    }

    BenefitFragment getFragment(int position) {
        return mFragments.get(position);
    }
}

File: app/src/main/kotlin/it/liceoarzignano/bold/intro/BenefitPageAdapter.kt
package it.liceoarzignano.bold.intro

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.util.SparseArray
import android.view.ViewGroup

internal class BenefitPageAdapter(manager: FragmentManager) : FragmentPagerAdapter(manager) {
    private val mFragments = SparseArray<BenefitFragment>()

    override fun getItem(position: Int): Fragment = BenefitFragment().newInstance(position)

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val fragment = super.instantiateItem(container, position) as BenefitFragment
        mFragments.put(position, fragment)
        return fragment
    }

    override fun getCount(): Int = 2

    val firstFragment: BenefitFragment
        get() = mFragments.get(0)
}